### ISSUE TITLE
[no-unused-vars] False positive enum

### ISSUE THREAD

**Issue by jorenvandeweyer** [typescript-eslint/typescript-eslint#2621]
- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
    "env": {
        "es2021": true,
        "node": true
    },
    "extends": [
        "standard"
    ],
    "parser": "@typescript-eslint/parser",
    "parserOptions": {
        "ecmaVersion": 12,
        "sourceType": "module"
    },
    "plugins": [
        "@typescript-eslint"
    ],
    "rules": {
    }
}
```

```TS
import { Column, Entity, PrimaryGeneratedColumn } from 'typeorm'

export enum UserRole {
  USER = 'user',
  ADMIN = 'admin',
}

@Entity()
export class User {
  @PrimaryGeneratedColumn()
  id: number;

  @Column({ type: 'enum', enum: UserRole, default: UserRole.USER })
  role: UserRole;
}
```

**Expected Result**
No error

**Actual Result**
`'USER' is defined but never used.eslintno-unused-vars`

**Additional Info**

The error also occurs without export and throws an additional error
`'UserRole' is defined but never used.eslintno-unused-vars`

```
eslint:cli CLI args: [ '--ext', 'ts', 'src/models/test.ts', '--debug' ] +0ms
  eslint:cli Running on files +5ms
  eslint:config-array-factory Loading JSON config file: /Users/joren/work/wilms/wilms-iot-api/package.json +0ms
  eslint:ignore-pattern Create with: [ IgnorePattern { patterns: [ '/**/node_modules/*' ], basePath: '/Users/joren/work/wilms/wilms-iot-api', loose: false } ] +0ms
  eslint:ignore-pattern   processed: { basePath: '/Users/joren/work/wilms/wilms-iot-api', patterns: [ '/**/node_modules/*' ] } +3ms
  eslint:ignore-pattern Create with: [ IgnorePattern { patterns: [ '/**/node_modules/*' ], basePath: '/Users/joren/work/wilms/wilms-iot-api', loose: false } ] +1ms
  eslint:ignore-pattern   processed: { basePath: '/Users/joren/work/wilms/wilms-iot-api', patterns: [ '/**/node_modules/*' ] } +1ms
  eslint:file-enumerator Start to iterate files: [ 'src/models/test.ts' ] +0ms
  eslint:file-enumerator File: /Users/joren/work/wilms/wilms-iot-api/src/models/test.ts +0ms
  eslint:cascading-config-array-factory Load config files for /Users/joren/work/wilms/wilms-iot-api/src/models. +0ms
  eslint:cascading-config-array-factory No cache found: /Users/joren/work/wilms/wilms-iot-api/src/models. +1ms
  eslint:config-array-factory Config file not found on /Users/joren/work/wilms/wilms-iot-api/src/models +9ms
  eslint:cascading-config-array-factory No cache found: /Users/joren/work/wilms/wilms-iot-api/src. +0ms
  eslint:config-array-factory Config file not found on /Users/joren/work/wilms/wilms-iot-api/src +0ms
  eslint:cascading-config-array-factory No cache found: /Users/joren/work/wilms/wilms-iot-api. +0ms
  eslint:config-array-factory Loading JSON config file: /Users/joren/work/wilms/wilms-iot-api/.eslintrc.json +1ms
  eslint:config-array-factory Config file found: /Users/joren/work/wilms/wilms-iot-api/.eslintrc.json +0ms
  eslint:config-array-factory Loading {extends:"standard"} relative to /Users/joren/work/wilms/wilms-iot-api/.eslintrc.json +0ms
  eslint:config-array-factory Loaded: eslint-config-standard@14.1.1 (/Users/joren/work/wilms/wilms-iot-api/node_modules/eslint-config-standard/index.js) +2ms
  eslint:config-array-factory Loading JS config file: /Users/joren/work/wilms/wilms-iot-api/node_modules/eslint-config-standard/index.js +0ms
  eslint:config-array-factory Loading plugin "import" from /Users/joren/work/wilms/wilms-iot-api/node_modules/eslint-config-standard/index.js +1ms
  eslint:config-array-factory Loaded: eslint-plugin-import@2.22.1 (/Users/joren/work/wilms/wilms-iot-api/node_modules/eslint-plugin-import/lib/index.js) +1ms
  eslint:config-array-factory Plugin /Users/joren/work/wilms/wilms-iot-api/node_modules/eslint-plugin-import/lib/index.js loaded in: 209ms +209ms
  eslint:config-array-factory Loading plugin "node" from /Users/joren/work/wilms/wilms-iot-api/node_modules/eslint-config-standard/index.js +0ms
  eslint:config-array-factory Loaded: eslint-plugin-node@11.1.0 (/Users/joren/work/wilms/wilms-iot-api/node_modules/eslint-plugin-node/lib/index.js) +1ms
  eslint:config-array-factory Plugin /Users/joren/work/wilms/wilms-iot-api/node_modules/eslint-plugin-node/lib/index.js loaded in: 59ms +59ms
  eslint:config-array-factory Loading plugin "promise" from /Users/joren/work/wilms/wilms-iot-api/node_modules/eslint-config-standard/index.js +0ms
  eslint:config-array-factory Loaded: eslint-plugin-promise@4.2.1 (/Users/joren/work/wilms/wilms-iot-api/node_modules/eslint-plugin-promise/index.js) +1ms
  eslint:config-array-factory Plugin /Users/joren/work/wilms/wilms-iot-api/node_modules/eslint-plugin-promise/index.js loaded in: 5ms +5ms
  eslint:config-array-factory Loading plugin "standard" from /Users/joren/work/wilms/wilms-iot-api/node_modules/eslint-config-standard/index.js +0ms
  eslint:config-array-factory Loaded: eslint-plugin-standard@4.0.1 (/Users/joren/work/wilms/wilms-iot-api/node_modules/eslint-plugin-standard/index.js) +0ms
  eslint:config-array-factory Plugin /Users/joren/work/wilms/wilms-iot-api/node_modules/eslint-plugin-standard/index.js loaded in: 1ms +1ms
  eslint:config-array-factory Loading parser "@typescript-eslint/parser" from /Users/joren/work/wilms/wilms-iot-api/.eslintrc.json +1ms
  eslint:config-array-factory Loaded: @typescript-eslint/parser@4.3.0 (/Users/joren/work/wilms/wilms-iot-api/node_modules/@typescript-eslint/parser/dist/index.js) +0ms
  eslint:config-array-factory Loading plugin "@typescript-eslint" from /Users/joren/work/wilms/wilms-iot-api/.eslintrc.json +222ms
  eslint:config-array-factory Loaded: @typescript-eslint/eslint-plugin@4.3.0 (/Users/joren/work/wilms/wilms-iot-api/node_modules/@typescript-eslint/eslint-plugin/dist/index.js) +1ms
  eslint:config-array-factory Plugin /Users/joren/work/wilms/wilms-iot-api/node_modules/@typescript-eslint/eslint-plugin/dist/index.js loaded in: 128ms +128ms
  eslint:cascading-config-array-factory No cache found: /Users/joren/work/wilms. +633ms
  eslint:config-array-factory Config file not found on /Users/joren/work/wilms +0ms
  eslint:cascading-config-array-factory No cache found: /Users/joren/work. +1ms
  eslint:config-array-factory Config file not found on /Users/joren/work +1ms
  eslint:cascading-config-array-factory No cache found: /Users/joren. +0ms
  eslint:cascading-config-array-factory Stop traversing because of considered root. +0ms
  eslint:rules Loading rule 'accessor-pairs' (remaining=281) +0ms
  eslint:rules Loading rule 'array-bracket-spacing' (remaining=280) +4ms
  eslint:rules Loading rule 'arrow-spacing' (remaining=279) +2ms
  eslint:rules Loading rule 'block-spacing' (remaining=278) +1ms
  eslint:rules Loading rule 'brace-style' (remaining=277) +1ms
  eslint:rules Loading rule 'camelcase' (remaining=276) +0ms
  eslint:rules Loading rule 'comma-dangle' (remaining=275) +2ms
  eslint:rules Loading rule 'comma-spacing' (remaining=274) +2ms
  eslint:rules Loading rule 'comma-style' (remaining=273) +1ms
  eslint:rules Loading rule 'computed-property-spacing' (remaining=272) +1ms
  eslint:rules Loading rule 'constructor-super' (remaining=271) +1ms
  eslint:rules Loading rule 'curly' (remaining=270) +3ms
  eslint:rules Loading rule 'dot-location' (remaining=269) +2ms
  eslint:rules Loading rule 'dot-notation' (remaining=268) +2ms
  eslint:rules Loading rule 'eol-last' (remaining=267) +2ms
  eslint:rules Loading rule 'eqeqeq' (remaining=266) +1ms
  eslint:rules Loading rule 'func-call-spacing' (remaining=265) +1ms
  eslint:rules Loading rule 'generator-star-spacing' (remaining=264) +1ms
  eslint:rules Loading rule 'handle-callback-err' (remaining=263) +3ms
  eslint:rules Loading rule 'indent' (remaining=262) +1ms
  eslint:rules Loading rule 'key-spacing' (remaining=261) +14ms
  eslint:rules Loading rule 'keyword-spacing' (remaining=260) +4ms
  eslint:rules Loading rule 'lines-between-class-members' (remaining=259) +18ms
  eslint:rules Loading rule 'new-cap' (remaining=258) +1ms
  eslint:rules Loading rule 'new-parens' (remaining=257) +4ms
  eslint:rules Loading rule 'no-array-constructor' (remaining=256) +2ms
  eslint:rules Loading rule 'no-async-promise-executor' (remaining=255) +0ms
  eslint:rules Loading rule 'no-caller' (remaining=254) +1ms
  eslint:rules Loading rule 'no-case-declarations' (remaining=253) +0ms
  eslint:rules Loading rule 'no-class-assign' (remaining=252) +1ms
  eslint:rules Loading rule 'no-compare-neg-zero' (remaining=251) +0ms
  eslint:rules Loading rule 'no-cond-assign' (remaining=250) +1ms
  eslint:rules Loading rule 'no-const-assign' (remaining=249) +0ms
  eslint:rules Loading rule 'no-constant-condition' (remaining=248) +1ms
  eslint:rules Loading rule 'no-control-regex' (remaining=247) +6ms
  eslint:rules Loading rule 'no-debugger' (remaining=246) +1ms
  eslint:rules Loading rule 'no-delete-var' (remaining=245) +1ms
  eslint:rules Loading rule 'no-dupe-args' (remaining=244) +0ms
  eslint:rules Loading rule 'no-dupe-class-members' (remaining=243) +2ms
  eslint:rules Loading rule 'no-dupe-keys' (remaining=242) +2ms
  eslint:rules Loading rule 'no-duplicate-case' (remaining=241) +0ms
  eslint:rules Loading rule 'no-empty-character-class' (remaining=240) +1ms
  eslint:rules Loading rule 'no-empty-pattern' (remaining=239) +1ms
  eslint:rules Loading rule 'no-eval' (remaining=238) +1ms
  eslint:rules Loading rule 'no-ex-assign' (remaining=237) +1ms
  eslint:rules Loading rule 'no-extend-native' (remaining=236) +0ms
  eslint:rules Loading rule 'no-extra-bind' (remaining=235) +6ms
  eslint:rules Loading rule 'no-extra-boolean-cast' (remaining=234) +1ms
  eslint:rules Loading rule 'no-extra-parens' (remaining=233) +1ms
  eslint:rules Loading rule 'no-fallthrough' (remaining=232) +2ms
  eslint:rules Loading rule 'no-floating-decimal' (remaining=231) +3ms
  eslint:rules Loading rule 'no-func-assign' (remaining=230) +2ms
  eslint:rules Loading rule 'no-global-assign' (remaining=229) +1ms
  eslint:rules Loading rule 'no-implied-eval' (remaining=228) +0ms
  eslint:rules Loading rule 'no-inner-declarations' (remaining=227) +1ms
  eslint:rules Loading rule 'no-invalid-regexp' (remaining=226) +0ms
  eslint:rules Loading rule 'no-irregular-whitespace' (remaining=225) +1ms
  eslint:rules Loading rule 'no-iterator' (remaining=224) +2ms
  eslint:rules Loading rule 'no-labels' (remaining=223) +2ms
  eslint:rules Loading rule 'no-lone-blocks' (remaining=222) +4ms
  eslint:rules Loading rule 'no-misleading-character-class' (remaining=221) +1ms
  eslint:rules Loading rule 'no-prototype-builtins' (remaining=220) +2ms
  eslint:rules Loading rule 'no-useless-catch' (remaining=219) +0ms
  eslint:rules Loading rule 'no-mixed-operators' (remaining=218) +0ms
  eslint:rules Loading rule 'no-mixed-spaces-and-tabs' (remaining=217) +2ms
  eslint:rules Loading rule 'no-multi-spaces' (remaining=216) +0ms
  eslint:rules Loading rule 'no-multi-str' (remaining=215) +1ms
  eslint:rules Loading rule 'no-multiple-empty-lines' (remaining=214) +1ms
  eslint:rules Loading rule 'no-negated-in-lhs' (remaining=213) +5ms
  eslint:rules Loading rule 'no-new' (remaining=212) +1ms
  eslint:rules Loading rule 'no-new-func' (remaining=211) +0ms
  eslint:rules Loading rule 'no-new-object' (remaining=210) +0ms
  eslint:rules Loading rule 'no-new-require' (remaining=209) +1ms
  eslint:rules Loading rule 'no-new-symbol' (remaining=208) +0ms
  eslint:rules Loading rule 'no-new-wrappers' (remaining=207) +0ms
  eslint:rules Loading rule 'no-obj-calls' (remaining=206) +1ms
  eslint:rules Loading rule 'no-octal' (remaining=205) +0ms
  eslint:rules Loading rule 'no-octal-escape' (remaining=204) +2ms
  eslint:rules Loading rule 'no-path-concat' (remaining=203) +0ms
  eslint:rules Loading rule 'no-proto' (remaining=202) +2ms
  eslint:rules Loading rule 'no-redeclare' (remaining=201) +2ms
  eslint:rules Loading rule 'no-regex-spaces' (remaining=200) +1ms
  eslint:rules Loading rule 'no-return-assign' (remaining=199) +1ms
  eslint:rules Loading rule 'no-self-assign' (remaining=198) +0ms
  eslint:rules Loading rule 'no-self-compare' (remaining=197) +1ms
  eslint:rules Loading rule 'no-sequences' (remaining=196) +1ms
  eslint:rules Loading rule 'no-shadow-restricted-names' (remaining=195) +0ms
  eslint:rules Loading rule 'no-sparse-arrays' (remaining=194) +6ms
  eslint:rules Loading rule 'no-tabs' (remaining=193) +1ms
  eslint:rules Loading rule 'no-template-curly-in-string' (remaining=192) +2ms
  eslint:rules Loading rule 'no-this-before-super' (remaining=191) +2ms
  eslint:rules Loading rule 'no-throw-literal' (remaining=190) +1ms
  eslint:rules Loading rule 'no-trailing-spaces' (remaining=189) +1ms
  eslint:rules Loading rule 'no-undef' (remaining=188) +2ms
  eslint:rules Loading rule 'no-undef-init' (remaining=187) +1ms
  eslint:rules Loading rule 'no-unexpected-multiline' (remaining=186) +1ms
  eslint:rules Loading rule 'no-unmodified-loop-condition' (remaining=185) +0ms
  eslint:rules Loading rule 'no-unneeded-ternary' (remaining=184) +1ms
  eslint:rules Loading rule 'no-unreachable' (remaining=183) +5ms
  eslint:rules Loading rule 'no-unsafe-finally' (remaining=182) +1ms
  eslint:rules Loading rule 'no-unsafe-negation' (remaining=181) +0ms
  eslint:rules Loading rule 'no-unused-expressions' (remaining=180) +2ms
  eslint:rules Loading rule 'no-unused-vars' (remaining=179) +3ms
  eslint:rules Loading rule 'no-use-before-define' (remaining=178) +3ms
  eslint:rules Loading rule 'no-useless-call' (remaining=177) +1ms
  eslint:rules Loading rule 'no-useless-computed-key' (remaining=176) +1ms
  eslint:rules Loading rule 'no-useless-constructor' (remaining=175) +1ms
  eslint:rules Loading rule 'no-useless-escape' (remaining=174) +0ms
  eslint:rules Loading rule 'no-useless-rename' (remaining=173) +1ms
  eslint:rules Loading rule 'no-useless-return' (remaining=172) +1ms
  eslint:rules Loading rule 'no-void' (remaining=171) +4ms
  eslint:rules Loading rule 'no-whitespace-before-property' (remaining=170) +2ms
  eslint:rules Loading rule 'no-with' (remaining=169) +1ms
  eslint:rules Loading rule 'object-curly-newline' (remaining=168) +0ms
  eslint:rules Loading rule 'object-curly-spacing' (remaining=167) +5ms
  eslint:rules Loading rule 'object-property-newline' (remaining=166) +2ms
  eslint:rules Loading rule 'one-var' (remaining=165) +2ms
  eslint:rules Loading rule 'operator-linebreak' (remaining=164) +2ms
  eslint:rules Loading rule 'padded-blocks' (remaining=163) +1ms
  eslint:rules Loading rule 'prefer-const' (remaining=162) +3ms
  eslint:rules Loading rule 'prefer-promise-reject-errors' (remaining=161) +4ms
  eslint:rules Loading rule 'quote-props' (remaining=160) +1ms
  eslint:rules Loading rule 'quotes' (remaining=159) +2ms
  eslint:rules Loading rule 'rest-spread-spacing' (remaining=158) +1ms
  eslint:rules Loading rule 'semi' (remaining=157) +1ms
  eslint:rules Loading rule 'semi-spacing' (remaining=156) +2ms
  eslint:rules Loading rule 'space-before-blocks' (remaining=155) +1ms
  eslint:rules Loading rule 'space-before-function-paren' (remaining=154) +1ms
  eslint:rules Loading rule 'space-in-parens' (remaining=153) +1ms
  eslint:rules Loading rule 'space-infix-ops' (remaining=152) +3ms
  eslint:rules Loading rule 'space-unary-ops' (remaining=151) +3ms
  eslint:rules Loading rule 'spaced-comment' (remaining=150) +2ms
  eslint:rules Loading rule 'symbol-description' (remaining=149) +3ms
  eslint:rules Loading rule 'template-curly-spacing' (remaining=148) +0ms
  eslint:rules Loading rule 'template-tag-spacing' (remaining=147) +1ms
  eslint:rules Loading rule 'unicode-bom' (remaining=146) +0ms
  eslint:rules Loading rule 'use-isnan' (remaining=145) +1ms
  eslint:rules Loading rule 'valid-typeof' (remaining=144) +1ms
  eslint:rules Loading rule 'wrap-iife' (remaining=143) +1ms
  eslint:rules Loading rule 'yield-star-spacing' (remaining=142) +1ms
  eslint:rules Loading rule 'yoda' (remaining=141) +1ms
  eslint:cascading-config-array-factory Configuration was determined: ConfigArray(3) [ { type: 'config', name: 'DefaultIgnorePattern', filePath: '', criteria: null, env: undefined, globals: undefined, ignorePattern: IgnorePattern { patterns: [Array], basePath: '/Users/joren/work/wilms/wilms-iot-api', loose: false }, noInlineConfig: undefined, parser: undefined, parserOptions: undefined, plugins: undefined, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: undefined, settings: undefined }, { type: 'config', name: '.eslintrc.json » eslint-config-standard', filePath: '/Users/joren/work/wilms/wilms-iot-api/node_modules/eslint-config-standard/index.js', criteria: null, env: { es6: true, node: true }, globals: { document: 'readonly', navigator: 'readonly', window: 'readonly' }, ignorePattern: undefined, noInlineConfig: undefined, parser: undefined, parserOptions: { ecmaVersion: 2020, ecmaFeatures: [Object], sourceType: 'module' }, plugins: { import: [Object], node: [Object], promise: [Object], standard: [Object] }, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: { 'accessor-pairs': 'error', 'array-bracket-spacing': [Array], 'arrow-spacing': [Array], 'block-spacing': [Array], 'brace-style': [Array], camelcase: [Array], 'comma-dangle': [Array], 'comma-spacing': [Array], 'comma-style': [Array], 'computed-property-spacing': [Array], 'constructor-super': 'error', curly: [Array], 'dot-location': [Array], 'dot-notation': [Array], 'eol-last': 'error', eqeqeq: [Array], 'func-call-spacing': [Array], 'generator-star-spacing': [Array], 'handle-callback-err': [Array], indent: [Array], 'key-spacing': [Array], 'keyword-spacing': [Array], 'lines-between-class-members': [Array], 'new-cap': [Array], 'new-parens': 'error', 'no-array-constructor': 'error', 'no-async-promise-executor': 'error', 'no-caller': 'error', 'no-case-declarations': 'error', 'no-class-assign': 'error', 'no-compare-neg-zero': 'error', 'no-cond-assign': 'error', 'no-const-assign': 'error', 'no-constant-condition': [Array], 'no-control-regex': 'error', 'no-debugger': 'error', 'no-delete-var': 'error', 'no-dupe-args': 'error', 'no-dupe-class-members': 'error', 'no-dupe-keys': 'error', 'no-duplicate-case': 'error', 'no-empty-character-class': 'error', 'no-empty-pattern': 'error', 'no-eval': 'error', 'no-ex-assign': 'error', 'no-extend-native': 'error', 'no-extra-bind': 'error', 'no-extra-boolean-cast': 'error', 'no-extra-parens': [Array], 'no-fallthrough': 'error', 'no-floating-decimal': 'error', 'no-func-assign': 'error', 'no-global-assign': 'error', 'no-implied-eval': 'error', 'no-inner-declarations': [Array], 'no-invalid-regexp': 'error', 'no-irregular-whitespace': 'error', 'no-iterator': 'error', 'no-labels': [Array], 'no-lone-blocks': 'error', 'no-misleading-character-class': 'error', 'no-prototype-builtins': 'error', 'no-useless-catch': 'error', 'no-mixed-operators': [Array], 'no-mixed-spaces-and-tabs': 'error', 'no-multi-spaces': 'error', 'no-multi-str': 'error', 'no-multiple-empty-lines': [Array], 'no-negated-in-lhs': 'error', 'no-new': 'error', 'no-new-func': 'error', 'no-new-object': 'error', 'no-new-require': 'error', 'no-new-symbol': 'error', 'no-new-wrappers': 'error', 'no-obj-calls': 'error', 'no-octal': 'error', 'no-octal-escape': 'error', 'no-path-concat': 'error', 'no-proto': 'error', 'no-redeclare': [Array], 'no-regex-spaces': 'error', 'no-return-assign': [Array], 'no-self-assign': [Array], 'no-self-compare': 'error', 'no-sequences': 'error', 'no-shadow-restricted-names': 'error', 'no-sparse-arrays': 'error', 'no-tabs': 'error', 'no-template-curly-in-string': 'error', 'no-this-before-super': 'error', 'no-throw-literal': 'error', 'no-trailing-spaces': 'error', 'no-undef': 'error', 'no-undef-init': 'error', 'no-unexpected-multiline': 'error', 'no-unmodified-loop-condition': 'error', 'no-unneeded-ternary': [Array], 'no-unreachable': 'error', 'no-unsafe-finally': 'error', 'no-unsafe-negation': 'error', 'no-unused-expressions': [Array], 'no-unused-vars': [Array], 'no-use-before-define': [Array], 'no-useless-call': 'error', 'no-useless-computed-key': 'error', 'no-useless-constructor': 'error', 'no-useless-escape': 'error', 'no-useless-rename': 'error', 'no-useless-return': 'error', 'no-void': 'error', 'no-whitespace-before-property': 'error', 'no-with': 'error', 'object-curly-newline': [Array], 'object-curly-spacing': [Array], 'object-property-newline': [Array], 'one-var': [Array], 'operator-linebreak': [Array], 'padded-blocks': [Array], 'prefer-const': [Array], 'prefer-promise-reject-errors': 'error', 'quote-props': [Array], quotes: [Array], 'rest-spread-spacing': [Array], semi: [Array], 'semi-spacing': [Array], 'space-before-blocks': [Array], 'space-before-function-paren': [Array], 'space-in-parens': [Array], 'space-infix-ops': 'error', 'space-unary-ops': [Array], 'spaced-comment': [Array], 'symbol-description': 'error', 'template-curly-spacing': [Array], 'template-tag-spacing': [Array], 'unicode-bom': [Array], 'use-isnan': 'error', 'valid-typeof': [Array], 'wrap-iife': [Array], 'yield-star-spacing': [Array], yoda: [Array], 'import/export': 'error', 'import/first': 'error', 'import/no-absolute-path': [Array], 'import/no-duplicates': 'error', 'import/no-named-default': 'error', 'import/no-webpack-loader-syntax': 'error', 'node/no-deprecated-api': 'error', 'node/process-exit-as-throw': 'error', 'promise/param-names': 'error', 'standard/no-callback-literal': 'error' }, settings: undefined }, { type: 'config', name: '.eslintrc.json', filePath: '/Users/joren/work/wilms/wilms-iot-api/.eslintrc.json', criteria: null, env: { es2021: true, node: true }, globals: undefined, ignorePattern: undefined, noInlineConfig: undefined, parser: { error: null, filePath: '/Users/joren/work/wilms/wilms-iot-api/node_modules/@typescript-eslint/parser/dist/index.js', id: '@typescript-eslint/parser', importerName: '.eslintrc.json', importerPath: '/Users/joren/work/wilms/wilms-iot-api/.eslintrc.json' }, parserOptions: { ecmaVersion: 12, sourceType: 'module' }, plugins: { '@typescript-eslint': [Object] }, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: {}, settings: undefined } ] on /Users/joren/work/wilms/wilms-iot-api/src/models +241ms
  eslint:ignore-pattern Create with: [ IgnorePattern { patterns: [ '/**/node_modules/*' ], basePath: '/Users/joren/work/wilms/wilms-iot-api', loose: false } ] +886ms
  eslint:ignore-pattern   processed: { basePath: '/Users/joren/work/wilms/wilms-iot-api', patterns: [ '/**/node_modules/*' ] } +0ms
  eslint:ignore-pattern Check {
  filePath: '/Users/joren/work/wilms/wilms-iot-api/src/models/test.ts',
  dot: false,
  relativePath: 'src/models/test.ts',
  result: false
} +1ms
  eslint:cli-engine Lint /Users/joren/work/wilms/wilms-iot-api/src/models/test.ts +0ms
  eslint:linter Linting code for /Users/joren/work/wilms/wilms-iot-api/src/models/test.ts (pass 1) +0ms
  eslint:linter Verify +0ms
  eslint:linter With ConfigArray: /Users/joren/work/wilms/wilms-iot-api/src/models/test.ts +1ms
  eslint:linter Generating fixed text for /Users/joren/work/wilms/wilms-iot-api/src/models/test.ts (pass 1) +106ms
  eslint:source-code-fixer Applying fixes +0ms
  eslint:source-code-fixer shouldFix parameter was false, not attempting fixes +0ms
  eslint:file-enumerator Complete iterating files: ["src/models/test.ts"] +995ms
  eslint:cli-engine Linting complete in: 996ms +108ms

/Users/joren/work/wilms/wilms-iot-api/src/models/test.ts
  5:3  error  'USER' is defined but never used       no-unused-vars
  6:3  error  'ADMIN' is defined but never used      no-unused-vars
  7:3  error  'INSTALLER' is defined but never used  no-unused-vars

✖ 3 problems (3 errors, 0 warnings)
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.3.0` |
| `@typescript-eslint/parser`        | `4.3.0` |
| `TypeScript`                       | `4.0.3` |
| `ESLint`                           | `7.10.0` |
| `node`                             | `14.6.0` |
| `typeorm` | `0.2.28` |

**Comment by yeonjuan** [comment]
@jorenvandeweyer 
seems duplicate of #2619. See [FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md#i-am-using-a-rule-from-eslint-core-and-it-doesnt-work-correctly-with-typescript-code), 
You should turn off the eslint core rule and turn on the typescript-eslint rule
```
rules: {
  "no-unused-vars": "off",
  "@typescript-eslint/no-unused-vars": "error"
}
```
